src/plugins.ts

```diff
@@ -105,23 +105,47 @@ async function loadPlugin(server: PluginsServer, pluginConfig: PluginConfig): Pr
     try {
         if (plugin.url?.startsWith('file:')) {
             const pluginPath = path.resolve(server.BASE_DIR, plugin.url.substring(5))
-            const configPath = path.resolve(pluginPath, 'plugin.json')
 
             let config: PluginJsonConfig = {}
-            if (fs.existsSync(configPath)) {
+            let wasConfigLoaded = false
+
+            for (const jsonFileName of ['package.json', 'plugin.json']) {
+                const configPath = path.resolve(pluginPath, jsonFileName)
                 try {
-                    const jsonBuffer = fs.readFileSync(configPath)
-                    config = JSON.parse(jsonBuffer.toString())
-                } catch (e) {
-                    await processError(
-                        server,
-                        pluginConfig,
-                        `Could not load posthog config at "${configPath}" for plugin "${plugin.name}"`
-                    )
-                    return false
+                    const rawConfigUpdate = fs.readFileSync(configPath).toString()
+                    let configUpdate: any
+                    try {
+                        configUpdate = JSON.parse(rawConfigUpdate)
+                    } catch {
+                        await processError(server, pluginConfig, `File ${jsonFileName} is not valid JSON`)
+                        return false
+                    }
+                    if (configUpdate['posthog-plugin-config']) {
+                        // plugin.json has `config`, but in package.json we prefer `posthog-plugin-config` for clarity
+                        configUpdate.config = configUpdate['posthog-plugin-config']
+                        delete configUpdate['posthog-plugin-config']
```

Comment:
Same stuff here, we should load `configSchema` instead of `posthog-plugin-config`.

Also, I think instead of this update system, it's probably fine to just 1) read plugin.json, 2) if not found, read package.json.